app/controllers/api/v1/display_ads_controller.rb

```diff
@@ -15,11 +15,40 @@ def show
         render json: @display_ad
       end
 
+      def create
+        @display_ad = DisplayAd.new(permitted_params)
+        result = @display_ad.save
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      rescue ArgumentError => e
+        # enums raise ArgumentError exceptions on unexpected inputs!
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def update
+        @display_ad = DisplayAd.find(params[:id])
+        result = @display_ad.update(permitted_params)
+        render json: @display_ad, status: (result ? :ok : :unprocessable_entity)
+      rescue ArgumentError => e
+        # enums raise ArgumentError exceptions on unexpected inputs!
+        render json: { error: e }, status: :unprocessable_entity
+      end
+
+      def destroy
```

Comment:
**thought**: I think it's misleading for a method named "destroy" to soft-delete instead. It's alright to soft-delete and we already do the same with Articles's API https://developers.forem.com/api/v1#tag/articles/operation/unpublishArticle, where we allow "unpublish" but not a full delete

Reply:
this actually got me curious again on how to keep controller as restful as possible and this makes me change my opinion on "destroy" a bit. Just an idea here, `DisplayAds::PublicationsController` with only `[:update, :destroy]` seems suitable.

Reply:
That's a good point, I'll match what we have in `articles#unpublish`.